test/setup/wdio.conf.ts

```diff
@@ -0,0 +1,142 @@
+/* eslint camelcase: ["error", {properties: "never"}] */
+/**
+ * See also: http://webdriver.io/guide/testrunner/configurationfile.html
+ */
+
+import { Options } from '@wdio/types';
+import { existsSync } from 'fs';
+import { dirname } from 'path';
+import { fileURLToPath } from 'url';
+import type { Capabilities } from '@wdio/types';
+import JsonReporter from '../helpers/json-reporter.js';
+import TestEnv from './TestEnv.js';
+
+// eslint-disable-next-line no-underscore-dangle
+const __dirname = dirname( fileURLToPath( import.meta.url ) );
+
+export function wdioConfig( providedTestEnv: TestEnv ): WebdriverIO.Config {
+	global.testEnv = providedTestEnv;
+	const settings = testEnv.settings;
+
+	return {
+		specs: settings.specs.map( ( specFilepath ) => `${__dirname}/../${specFilepath}` ),
+
+		hostname: 'browser',
+		port: 4444,
+		path: '/wd/hub',
+		// ============
+		// Capabilities
+		// ============
+		// https://sites.google.com/a/chromium.org/chromedriver/capabilities
+		capabilities: [
+			{
+				browserName: 'chrome',
+				maxInstances: settings.maxInstances,
+				'goog:chromeOptions': {
+					args: [
+						// The window size is relevant for responsive pages rendering differently on
+						// different screen sizes. Bootstrap considers widths between 1200 and 1400
+						// as XL, let's use that.
+						// https://getbootstrap.com/docs/5.0/layout/breakpoints/#available-breakpoints
+						'--window-size=1280,800',
+						// https://www.selenium.dev/blog/2023/headless-is-going-away
+						...( settings.runHeaded ? [] : [ '--headless=new' ] ),
+						// Chrome sandbox does not work in Docker
+						...( existsSync( '/.dockerenv' ) ? [ '--no-sandbox' ] : [] )
+					]
+				}
+			} as Capabilities.ChromeCapabilities
+		],
+
+		// ===================
+		// Test Configurations
+		// ===================
+
+		// Level of verbosity: "trace", "debug", "info", "warn", "error", "silent"
+		logLevel: settings.logLevel as Options.WebDriverLogTypes,
+
+		outputDir: settings.outputDir,
+
+		// Default timeout for each waitFor* command.
+		waitforTimeout: settings.waitForTimeout as number,
```

Comment:
Can we either make `waitForTimeout` non-optional or set a default here?

Reply:
The default values for this are set in `makeTestSettings` and come from `test-runner.env` which is equally true for `testTimeout`, `logLevel` and a few others in here. `wdio.conf.ts` is a bit of a shell at this point, and on purpose. I've added the comment to that effect from my note in this PR  to the top of this file to make that more clear.  Also have now made this and all other `TestRunnerSettings` properties required (except `runHeaded` which is truly optional.